BaseCollector.py

```diff
@@ -113,21 +113,28 @@ def wait_for_inventory_data(self):
         print("done: initial query " + type(self).__name__)
         return
 
-    def generate_gauges(self, metric_type, calling_class, vrops_entity_name, labelnames):
+    def generate_gauges(self, metric_type, calling_class, vrops_entity_name, labelnames, rubric=None):
         if not isinstance(labelnames, list):
             print("Can't generate Gauges without label list, called from", calling_class)
             return {}
         # switching between metric and property types
         if metric_type == 'stats':
             statkey_yaml = self.read_collector_config()['statkeys']
             gauges = dict()
-            for statkey_pair in statkey_yaml[calling_class]:
+
+            def iterate():
                 statkey_suffix = statkey_pair['metric_suffix']
                 gauges[statkey_suffix] = {
                     'gauge': GaugeMetricFamily('vrops_' + vrops_entity_name + '_' + statkey_suffix.lower(),
                                                'vrops-exporter', labels=labelnames),
                     'statkey': statkey_pair['statkey']
                 }
+            if calling_class == 'VMStatsCollector':
+                for statkey_pair in statkey_yaml[calling_class][rubric]:
+                    iterate()
```

Comment:
wouldn't it be better to define a method BaseCollector.generateGaugesByStatkey() and return gauges?

Reply:
I solved it by checking, if `rubrics=True`.